### PR TITLE
Add email diagnostics control panel

### PR DESCRIPTION
**Pull request by alvarosanchez**
## Summary

- add `micronaut-control-panel-email` with grouped Micronaut Email sender diagnostics, provider inference, safe configuration checklist data, and an empty-state panel
- add a guarded, disabled-by-default test-send helper route with configured-recipient defaults and redacted success/failure responses
- document the email panel and wire a no-op sender into the Java example for UI verification

## Issue Linkage

Paperclip issue: DEV-484.

No linked GitHub issue was found during QA intake, Security review, or Code review, so no GitHub closing keyword or issue-creator reviewer is applicable.

## Organization Projects

Live Micronaut organization project association:

- `5.1.0 Release` (#147)

Target/project note: QA originally selected `5.0.0-M3` (#143) and `5.0.0 Release` (#47) for the earlier `2.0.x` target with a Control Panel release-line ambiguity note. PR #564 is now retargeted to `2.1.x`, and the live PR association is `5.1.0 Release`.

## Screenshots

Desktop light, 1440x900:

Desktop light

Desktop dark, 1440x900:

Desktop dark

Mobile dark, 390x844:

Mobile dark

## Testing

- `./gradlew :micronaut-control-panel-email:check :micronaut-doc-examples:micronaut-example-java:test --tests example.ControlPanelRenderingTest --continue`
- `./gradlew docs`
- `git diff --check`

---
###### ✨ This message was AI-generated using gpt-5

### CODE REVIEW

File: control-panel-email/src/main/java/io/micronaut/controlpanel/panels/email/EmailController.java [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+        if (!testSend.isEnabled()) {
+            return HttpResponse.badRequest(TestSendResult.disabled("Test send is disabled.", elapsed(started)));
+        }
+        String recipient = recipient(request);
+        if (StringUtils.isEmpty(recipient)) {
+            return HttpResponse.badRequest(TestSendResult.disabled("No safe test recipient is configured.", elapsed(started)));
+        }
+        if (request != null && StringUtils.isNotEmpty(request.recipient()) && !testSend.isAllowArbitraryRecipient()) {
+            return HttpResponse.badRequest(TestSendResult.disabled("Arbitrary recipients are disabled for test send.", elapsed(started)));
+        }
```
[the remark this answers is not in the document]
Comment (alvarosanchez) [reply control-panel-email/src/main/java/io/micronaut/controlpanel/panels/email/EmailController.java]: Committed the requested change in 7320816. The controller now requires a configured safe test recipient before it will honor any request-provided recipient, even when arbitrary recipients are enabled. I also added the regression test for enabled test send, empty configured recipient, arbitrary recipient allowed, and request recipient present returning 400 with no sender invocation.

---
###### ✨ This message was AI-generated using gpt-5

**Comment by alvarosanchez** [comment]
Additional coverage follow-up pushed in 235c981 after the 02410f3 Sonar run cleared all issues but still failed new-code coverage at 79.7%.

This commit adds focused tests for existing email panel getters, empty-state view metadata, model helper methods, and configuration accessors. No production behavior changed.

Local verification passed again:
- `./gradlew :micronaut-control-panel-email:check`
- `git diff --check`

---
###### ✨ This message was AI-generated using gpt-5

**Comment by alvarosanchez** [comment]
Retargeted this PR from `2.1.x` to `2.2.x`, and merged the new base in.

Rationale: this PR adds a new module (`control-panel-email`), and `2.1.x` has shipped `v2.1.0` and is now `2.1.1-SNAPSHOT`, a patch line. `CONTRIBUTING.md` ("Merging a pull request") puts a new module into the next minor version, and `2.2.x` (`2.2.0-SNAPSHOT`) is now the default branch. `2.1.x` is fully contained in `2.2.x`, so no content was lost and the feature diff is unchanged.

Two build-level adjustments were needed for the new base, no production source/test/doc changes:

1. `gradle/libs.versions.toml` — kept `micronaut-email = "3.0.0-RC1"`, took the base's `micronaut-openapi = "7.1.2"` and `micronaut-kafka = "6.1.0"`.
2. `control-panel-email/build.gradle.kts` — `binaryCompatibility.enabledAfter("2.1.1")` → `("2.2.0")`. Moving to `2.2.0-SNAPSHOT` turned the check on, and `findBaseline` then failed with `Could not find a previous version for 2.2.0` because this module has never been published. With `"2.2.0"` the check is correctly skipped for the first release of a new module.

Verified locally: `./gradlew :micronaut-control-panel-email:check` passes (15/15 tests).

The organization project (`5.2.0 Release`) and the `type: enhancement` label are unchanged.

---
###### ✨ This message was AI-generated using claude-opus-5
